pkg/kubelet/kubelet.go

```diff
@@ -2813,6 +2824,11 @@ func (kl *Kubelet) setNodeStatusMachineInfo(node *api.Node) {
 				"Node %s has been rebooted, boot id: %s", kl.nodeName, info.BootID)
 		}
 		node.Status.NodeInfo.BootID = info.BootID
+
+		// If no cloud provider is defined - use the one detected by cadvisor
+		if kl.cloud == nil {
+			kl.updateCloudProviderFromMachineInfo(node, info)
```

Comment:
Should this ever change?  If not, I'd like it in `initialNodeStatus` so that we aren't generating changes that will be ignored.